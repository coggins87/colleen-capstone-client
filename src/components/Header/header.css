nav {
  position: fixed;
  z-index: 1;
  background-color: rgb(46, 47, 48);
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-self: flex-start;
  width: 100%;
}
h1 {
  font-family: "Nanum Myeongjo", serif;
}
.Header__tagline--wide {
  color: rgb(199, 199, 199);
  margin-top: -5%;
  padding-bottom: 2%;
}
.header-div {
  justify-content: flex-start;
  flex-direction: inherit;
  margin: auto;
}

.page-links {
  flex-direction: column;
  justify-content: flex-end;
  text-align: right;
  margin: auto;
}

h1 > .link {
  color: rgb(220, 228, 222);
  font-size: 1.5rem;
}
.link {
  text-decoration: none;
  color: whitesmoke;
}

.link:hover {
  color: rgb(0, 183, 255);
}

@media all and (max-width: 1100px) {
  nav {
    align-content: center;
    width: 100%;
  }
}
@media all and (max-width: 800px) {
  nav {
    align-content: center;
    width: 100%;
  }
}
@media all and (max-width: 510px) {
  nav {
    flex-direction: column;
    align-content: center;
    width: 100%;
  }

  h1 {
    text-align: center;
  }
  .page-links {
    justify-content: left;
    text-align: center;
  }
  .Header__not-logged-in {
    text-align: center;
  }
  .Header__logged-in {
    text-align: center;
  }
  .link:click {
    color: rgb(0, 183, 255);
  }
}
