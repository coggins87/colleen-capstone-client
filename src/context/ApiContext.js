import React from "react";
import TokenService from "../services/token-service";
import WorkoutService from "../services/workout-search-service";
const ApiContext = React.createContext({
  error: null,
  isLoggedIn: false,
  userId: null,
  setError: () => {},
  clearError: () => {},
  handleLoginSuccess: () => {}
});

export default ApiContext;

export class ApiProvider extends React.Component {
  state = {
    error: null,
    isLoggedIn: false,
    userId: null
  };
  setError = error => {
    console.error(error);
    this.setState({ error });
  };
  clearError = () => {
    this.setState({ error: null });
  };

  onDelete = () => {
    const user_id = this.props.reqs.params.userId;
    WorkoutService.deleteWorkout(user_id);
    this.setState({ workouts: WorkoutService.getUserWorkouts(user_id) });
  };

  nullError = () => {
    this.setState( {error: 'Invalid user, close this window to log out and try again'})
  }

  handleLoginSuccess = () => {
    const read = TokenService.readJwtToken();
    const user_Id = read.user_id;
    this.setState({
      userId: user_Id,
      isLoggedIn: true
    });
  };

  handleLogoutSuccess = () => {
    this.setState({
      userId: null,
      isLoggedIn: false
    });
  };

  isUserLoggedIn = () => {
    if (TokenService.hasAuthToken()) {
      const read = TokenService.readJwtToken();
      const user_Id = read.user_id;
      this.setState({
        userId: user_Id
      });
    }
  };
  render() {
    const value = {
      isUserLoggedIn: this.isUserLoggedIn,
      nullError: this.nullError,
      error: this.state.error,
      isLoggedIn: this.state.isLoggedIn,
      setError: this.setError,
      clearError: this.clearError,
      userId: this.state.userId,
      handleLoginSuccess: this.handleLoginSuccess,
      handleLogoutSuccess: this.handleLogoutSuccess,
      handleDeleteWorkout: this.onDelete
    };
    return (
      <ApiContext.Provider value={value}>
        {this.props.children}
      </ApiContext.Provider>
    );
  }
}
