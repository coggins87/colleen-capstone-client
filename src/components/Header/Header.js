import React from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
import IdleService from '../../services/idle-service'
import ApiContext from '../../context/ApiContext';

class Header extends React.Component {

  state = {
    workouts: []
  };
  static defaultProps = {
    match: { params: {} }
  };

  static contextType = ApiContext

  handleLogoutClick = () => {
   TokenService.clearAuthToken()
   TokenService.clearCallbackBeforeExpiry()
   IdleService.unRegisterIdleResets()
    this.context.isLoggedIn = false
   this.forceUpdate()
  }

  renderLogoutLink() {
    return (
      <div className='Header__logged-in'>
        <Link
          onClick={this.handleLogoutClick}
          to='/'>
          Logout
        </Link>
      </div>
    )
  }

  renderLoginLink() {
    return (
      <div className='Header__not-logged-in'>
        <Link
          to='/login'>
          Log in
        </Link>
        <Link
          to='/register'>
          Register
        </Link>
      </div>
    )
  }

  renderUserPageLink() {
    return (
      <div className='Header__logged-in'>
        <Link
        to='/users/:userId'>
          My Workouts
        </Link>
      </div>
    )
  }

  render(){
    return(
  <nav>
    <h1>
    <Link to='/'>
      AMRApp
    </Link>
    </h1>
    <span className='Header__tagline--wide'>Find your next workout!</span>
    
    {TokenService.hasAuthToken()|| this.context.isLoggedIn ? this.renderLogoutLink() :
    this.renderLoginLink()}
    {TokenService.hasAuthToken() || this.context.isLoggedIn ? this.renderUserPageLink() : <></>}
    <Link to='/search'>Generate A Workout</Link> 
    
  </nav>
  )
  }
}

export default Header