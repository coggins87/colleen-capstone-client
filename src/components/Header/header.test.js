import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Header from "./Header";
import { ApiProvider } from "../../context/ApiContext";

describe(`Header component`, () => {
  const props = {
    testObj: {},
    history: {
      testCheck: () => {}
    }
  };

  it("renders a Header by default", () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the Header given props", () => {
    const wrapper = shallow(<Header {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the Header given context", () => {
    const wrapper = shallow(
      <ApiProvider>
        <Header />
      </ApiProvider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
