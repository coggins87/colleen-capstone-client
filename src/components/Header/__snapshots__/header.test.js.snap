// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders a Header by default 1`] = `
<nav>
  <div
    className="header-div"
  >
    <h1>
      <Link
        className="header, link"
        to="/"
      >
        AMRApp
      </Link>
    </h1>
  </div>
  <div
    className="page-links"
  >
    <div
      className="Header__not-logged-in"
    >
      <Link
        className="link"
        to="/users/login"
      >
        Log in
      </Link>
      <div>
        <Link
          className="link"
          to="/register"
        >
          Register
        </Link>
      </div>
    </div>
    <Link
      className="link"
      to="/search"
    >
      Generate A Workout
    </Link>
  </div>
</nav>
`;

exports[`Header component renders the Header given context 1`] = `
<ContextProvider
  value={
    Object {
      "clearError": [Function],
      "error": null,
      "handleDeleteWorkout": [Function],
      "handleLoginSuccess": [Function],
      "handleLogoutSuccess": [Function],
      "isLoggedIn": false,
      "isUserLoggedIn": [Function],
      "nullError": [Function],
      "setError": [Function],
      "userId": null,
    }
  }
>
  <Header
    history={
      Object {
        "push": [Function],
      }
    }
    location={Object {}}
  />
</ContextProvider>
`;

exports[`Header component renders the Header given props 1`] = `
<nav>
  <div
    className="header-div"
  >
    <h1>
      <Link
        className="header, link"
        to="/"
      >
        AMRApp
      </Link>
    </h1>
  </div>
  <div
    className="page-links"
  >
    <div
      className="Header__not-logged-in"
    >
      <Link
        className="link"
        to="/users/login"
      >
        Log in
      </Link>
      <div>
        <Link
          className="link"
          to="/register"
        >
          Register
        </Link>
      </div>
    </div>
    <Link
      className="link"
      to="/search"
    >
      Generate A Workout
    </Link>
  </div>
</nav>
`;
